Sources/Main.cpp

```diff
@@ -603,30 +600,60 @@ void __fastcall TX584Form::CodeListViewCustomDrawItem(
 }
 //---------------------------------------------------------------------------
 
+void __fastcall TX584Form::CodeListViewAdvancedCustomDrawItem(TCustomListView *Sender,
+          TListItem *Item, TCustomDrawState State, TCustomDrawStage Stage,
+          bool &DefaultDraw)
+{
+    DefaultDraw = false;
+    TRect Rectangle = Item->DisplayRect(drBounds);
+    TRect TextRectangle;
+
+    int LastColumnIndex = CodeListView->Columns->Count - 2;
+    for (int i = 0; i <= LastColumnIndex; i++) {
```

Comment:
The variable `i` here actually means index of **previous** column. This is a bit confusing. Usually the loop counter denotes the current item. What about more natural loop?

```cpp
for (int i = 1; i < CodeListView->Columns->Count; i++) {
```

This clearly says you're going to draw all columns except column 0.